cirq-aqt/cirq_aqt/aqt_sampler.py

```diff
@@ -99,12 +104,64 @@ def _generate_json(
             raise RuntimeError('Cannot send an empty circuit')
         json_str = json.dumps(seq_list)
         return json_str
+    
+    def _parse_legacy_circuit_json(self, json_str: str) -> list:
+        """Converts a legacy JSON circuit representation.
+        
+        Converts a JSON created for the legacy API into one that will work
+        with the new API.
+
+        Args:
+            json_str: A JSON-formatted string that could be used as the
+                data parameter in the body of a request to the old AQT API.
+        """
+        circuit = []
+        number_of_measurements = 0
+
+        for legacy_op in json.loads(json_str):
+            if number_of_measurements > 0:
+                raise ValueError(
+                    "Need exactly one `MEASURE` operation at the end of the circuit."
+                )
+
+            instruction = {}
+
+            match legacy_op[0]:
```

Comment:
Probably Cirq needs to support Python versions < 3.10

See: https://github.com/quantumlib/Cirq/blob/main/setup.py#L67C5-L67C31